### ISSUE TITLE
f5remark and f5label JSON schema

### ISSUE THREAD

**Issue by walterk82** [F5Networks/f5-appsvcs-extension#234]
### Environment
 * Application Services Version: 3.17.0
 * BIG-IP Version: 14.1.2.3

### Summary
The "label" and "remark" tags of the JSON schema do not provide length validation of 64 maximum characters.  This causes the declaration to validate in VS.Code, but fail to POST with an HTTP 422

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "AS3",
    "action": "deploy",
    "persist": false,
    "historyLimit": 2,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.17.0",
        "Tenant1": {
            "class": "Tenant",
            "App1": {
                "class": "Application",
                "template": "generic",
                "Vip1_443_vs": {
                    "class": "Service_HTTPS",
                    "remark": "This is my really long remark comment that should fail the 64 character check",
                    "virtualAddresses": ["10.1.1.1"],
                    "virtualPort": 443,
                    "redirect80": false,
                    "serverTLS": {"bigip": "/Common/clientssl"},
                    "clientTLS": {"bigip": "/Common/serverssl"},
                    "snat": "none",
                    "pool": "443_pool",
                    "persistenceMethods": []
                },
                "Vip1_443_pool": {
                    "class": "Pool",
                    "members": [{"servicePort": 443, "serverAddresses": ["10.1.2.2", "10.1.2.3"]}]
                }
            }
        }
    }
}
```

2. Observe the following error response:
```json
{
    "code": 422,
    "errors": [
        "/Tenant1/App1/Vip1_443_vs/remark: should match format \"f5remark\""
    ],
    "declarationFullId": "",
    "message": "declaration is invalid"
}
```

### Expected Behavior
as3-schema.json should be updated to enforce character length checking for "label" and "remark" according to https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/refguide/understanding-the-json-schema.html

**Comment by dstokesf5** [comment]
The AS3 schema is complex and makes use of several custom keywords and formats. Other tools will not understand those keywords and formats, and there are not a lot of options for us to make those tools understand those keywords and formats.

This is not something we can fix with a quick schema update.

**Comment by walterk82** [comment]
I see maxLength used in other areas of the schema, I don't understand why this is too complicated to implement.:
https://json-schema.org/understanding-json-schema/reference/string.html

**Comment by dstokesf5** [comment]
Those other areas of the schema are not using a custom format. The f5remark format is a custom one that is handled in code. In order to get VS code to understand those custom formats, we would have to replace all uses of that custom format in the schema with different schema that makes use of things like the pattern and maxLength keywords.

**Comment by walterk82** [comment]
What advantage does a custom format provide here?  Pointer usage?  Usability of the syntax is the biggest hindrance to adoption of AS3.

**Comment by walterk82** [comment]
I just modified the schema and length checking is working fine.

**Comment by dstokesf5** [comment]
The f5remark and f5label custom formats are used in a few places, and all places they are used in the schema will need to be updated. However, it looks like f5remark and f5label are not used in as many places as I initially thought. It looks like they are partially set up to be reusable in a way that is more compatible with other tools. I will get a task added into our internal product backlog for this, but it will likely be some time before we can get to it.

**Comment by walterk82** [comment]
I just added "maxLength" on line 7582 of the 3.17 schema.

**Comment by dstokesf5** [comment]
The two keywords appear to be used in at least 4 places throughout the schema. I have created AUTOTOOL-1243 to change them.

**Comment by walterk82** [comment]
@dstokesf5 Was this issue addressed?  I saw some notes about removing these custom types.

**Comment by dstokesf5** [comment]
Yes, it looks like this issue was resolved by the 3.27.0 release. Thank you for pointing this out.
